apps/server/src/modules/tool/mapper/external-tool-request.mapper.spec.ts

```diff
@@ -0,0 +1,210 @@
+		lti11ConfigParams.secret = 'mockSecret';
+		lti11ConfigParams.resource_link_id = 'mockLink';
+		lti11ConfigParams.lti_message_type = LtiMessageType.BASIC_LTI_LAUNCH_REQUEST;
+		lti11ConfigParams.privacy_permission = LtiPrivacyPermission.NAME;
+
+		const oauth2ConfigParams = new Oauth2ToolConfigParams();
+		oauth2ConfigParams.type = ToolConfigType.OAUTH2;
+		oauth2ConfigParams.baseUrl = 'mockUrl';
+		oauth2ConfigParams.clientId = 'mockId';
+		oauth2ConfigParams.clientSecret = 'mockSecret';
+		oauth2ConfigParams.frontchannelLogoutUri = 'mockUrl';
+		oauth2ConfigParams.skipConsent = true;
+		oauth2ConfigParams.scope = 'mockScope';
+		oauth2ConfigParams.redirectUris = ['mockUri'];
+		oauth2ConfigParams.tokenEndpointAuthMethod = TokenEndpointAuthMethod.CLIENT_SECRET_POST;
+
+		const customParameterCreateParams = new CustomParameterCreateParams();
+		customParameterCreateParams.name = 'mockName';
+		customParameterCreateParams.default = 'mockDefault';
+		customParameterCreateParams.location = CustomParameterLocationParams.PATH;
+		customParameterCreateParams.scope = CustomParameterScopeParams.SCHOOL;
+		customParameterCreateParams.type = CustomParameterTypeParams.STRING;
+		customParameterCreateParams.regex = 'mockRegex';
+
+		const externalToolParams = new ExternalToolParams();
+		externalToolParams.name = 'mockName';
+		externalToolParams.url = 'mockUrl';
+		externalToolParams.logoUrl = 'mockLogoUrl';
+		externalToolParams.parameters = [customParameterCreateParams];
+		externalToolParams.isHidden = true;
+		externalToolParams.openNewTab = true;
+
+		const basicToolConfigDO: BasicToolConfigDO = new BasicToolConfigDO({
+			type: ToolConfigType.BASIC,
+			baseUrl: 'mockUrl',
+		});
+		const lti11ToolConfigDO: Lti11ToolConfigDO = new Lti11ToolConfigDO({
+			privacy_permission: LtiPrivacyPermission.NAME,
+			secret: 'mockSecret',
+			key: 'mockKey',
+			resource_link_id: 'mockLink',
+			lti_message_type: LtiMessageType.BASIC_LTI_LAUNCH_REQUEST,
+			type: ToolConfigType.LTI11,
+			baseUrl: 'mockUrl',
+		});
+		const oauth2ToolConfigDO: Oauth2ToolConfigDO = new Oauth2ToolConfigDO({
+			clientId: 'mockId',
+			type: ToolConfigType.OAUTH2,
+			baseUrl: 'mockUrl',
+			clientSecret: 'mockSecret',
+			frontchannelLogoutUri: 'mockUrl',
+			skipConsent: true,
+			scope: 'mockScope',
+			redirectUris: ['mockUri'],
+			tokenEndpointAuthMethod: TokenEndpointAuthMethod.CLIENT_SECRET_POST,
+		});
+		const customParameterDO: CustomParameterDO = new CustomParameterDO({
+			name: 'mockName',
+			default: 'mockDefault',
+			location: CustomParameterLocation.PATH,
+			scope: CustomParameterScope.SCHOOL,
+			type: CustomParameterType.STRING,
+			regex: 'mockRegex',
+		});
+		const externalToolDO: ExternalToolDO = new ExternalToolDO({
+			name: 'mockName',
+			url: 'mockUrl',
+			logoUrl: 'mockLogoUrl',
+			parameters: [customParameterDO],
+			isHidden: true,
+			openNewTab: true,
+			version: 1,
+			config: basicToolConfigDO,
+		});
+
+		return {
+			externalToolParams,
```

Comment:
this might make sense to split into multiple setups/when clauses as well

Reply:
done